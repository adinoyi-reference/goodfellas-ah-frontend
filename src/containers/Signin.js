import React from 'react';

const Signin = () => (
  <div>
    <input type="text" />
    <input type="text" />
    <button type="button">Sign In</button>
  </div>
);

export default Signin;
