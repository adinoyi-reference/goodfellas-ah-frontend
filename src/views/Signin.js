import React from 'react';
import SigninForm from '../containers/SigninForm';
import Auth from '../components/layout/Auth';


const Signin = (props) => (
  <div>
    <Auth><SigninForm {...props} /></Auth>
  </div>
);

export default Signin;
