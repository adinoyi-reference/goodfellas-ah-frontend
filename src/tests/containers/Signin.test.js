import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Signin from '../../containers/SigninForm';
import Root from '../../root';

let wrapped;

beforeEach(() => {
  wrapped = mount(
    <Root>
      <Signin />
    </Root>,
  );
});

afterEach(() => wrapped.unmount());

describe('Signin UI', () => {
  describe('render features', () => {
    test('container should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });

    it('has two input fields and a button', () => {
      expect(wrapped.find('input').length).toEqual(2);
      expect(wrapped.find('button').length).toEqual(1);
    });
  });

  describe('when typing into fields', () => {
    beforeEach(() => {
      wrapped.find('input').first().simulate('change', { target: { id: 'email', value: 'user@email.com' } });
      wrapped.find('input').at(1).simulate('change', { target: { id: 'password', value: 'password' } });
      wrapped.update();
    });

    it('shows that text has been entered into then email field', () => {
      expect(wrapped.find('input').first().prop('value')).toEqual('user@email.com');
    });

    it('shows that text has been entered into then password field', () => {
      expect(wrapped.find('input').at(1).prop('value')).toEqual('password');
    });

    it('empties fields when form is submitted', () => {
      wrapped.find('form').simulate('submit');
      wrapped.update();

      expect(wrapped.find('input').first().prop('value')).toEqual('');
      expect(wrapped.find('input').at(1).prop('value')).toEqual('');
    });
  });
});

describe('Signin client-side validation', () => {
  beforeEach(() => {
    wrapped.find('input').first().simulate('change', { target: { id: 'email', value: '' } });
    wrapped.find('input').at(1).simulate('change', { target: { id: 'password', value: '' } });
    wrapped.update();
  });

  it('shows error messages if both fields are empty', () => {
    wrapped.find('form').simulate('submit');
    wrapped.update();

    expect(wrapped.find('.auth-input-wrapper').first().find('.error-field').text()).toEqual('email field must be filled');
    expect(wrapped.find('.auth-input-wrapper').at(1).find('.error-field').text()).toEqual('password field must be filled');
  });
});
