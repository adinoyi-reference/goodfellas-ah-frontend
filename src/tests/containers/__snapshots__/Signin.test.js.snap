// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin UI render features container should render as expected 1`] = `
<_default>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <MemoryRouter
      initialEntries={
        Array [
          Object {
            "key": "testKey",
          },
        ]
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "key": "testKey",
                "pathname": "/",
                "search": "",
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/",
              "search": "",
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Connect(Signin)>
          <Signin
            signin={[Function]}
          >
            <form
              className="auth-signin"
              onSubmit={[Function]}
            >
              <div
                className="error-field"
              />
              <AuthInput
                error={
                  Object {
                    "email": "Please enter a valid email.",
                    "password": "Please enter a valid password.",
                    "status": true,
                  }
                }
                handleBlur={[Function]}
                handleChange={[Function]}
                name="email"
                placeholder="Email"
                touched={
                  Object {
                    "email": false,
                    "password": false,
                  }
                }
                value=""
              >
                <div
                  className="auth-input-wrapper"
                >
                  <label
                    className="sr-only"
                    htmlFor="email"
                  >
                    email
                  </label>
                  <input
                    className="auth-field "
                    id="email"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Email"
                    type="text"
                    value=""
                  />
                  <div
                    className="error-field"
                  />
                </div>
              </AuthInput>
              <AuthInput
                error={
                  Object {
                    "email": "Please enter a valid email.",
                    "password": "Please enter a valid password.",
                    "status": true,
                  }
                }
                handleBlur={[Function]}
                handleChange={[Function]}
                name="password"
                placeholder="Password"
                touched={
                  Object {
                    "email": false,
                    "password": false,
                  }
                }
                type="password"
                value=""
              >
                <div
                  className="auth-input-wrapper"
                >
                  <label
                    className="sr-only"
                    htmlFor="password"
                  >
                    password
                  </label>
                  <input
                    className="auth-field "
                    id="password"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                  <div
                    className="error-field"
                  />
                </div>
              </AuthInput>
              <div>
                <Link
                  replace={false}
                  to="/forgotpassword"
                >
                  <a
                    href="/forgotpassword"
                    onClick={[Function]}
                  >
                    <div
                      className="forgot-password"
                    >
                      Forgot Password?
                    </div>
                  </a>
                </Link>
              </div>
              <Button
                className="auth-button"
                title="SIGN IN"
                type="submit"
              >
                <button
                  className="auth-button"
                  type="submit"
                >
                  SIGN IN
                </button>
              </Button>
            </form>
          </Signin>
        </Connect(Signin)>
      </Router>
    </MemoryRouter>
  </Provider>
</_default>
`;
